### ISSUE TITLE
Proposal: Shorter chapter/section names, possible reordering

### ISSUE THREAD

**Issue by Ayertienna** [Beluga-lang/Meta#3]
This is a very high-level suggestion, more a starting point of a discussion than anything else.

A couple of points:
- Right now the chapter and section names are a bit wordy
- It might make sense to add a level above "chapter" ("part"), since eventually there will be more than one advanced chapter
- In the sections about encoding proofs, we should mention each new proof technique/trick clearly in a section title (for people using this book more to learn how to do _their thing_ in Beluga)
- @Niols suggested to perhaps focus more on the introduced proof technique/Beluga feature rather than the specific proof itself in those sections (this is the ALT variant)

My proposal would therefore be along the lines of:
I The basics of encodings
1 Language of arithmetic and boolean expressions
(possibly these should be paragraphs rather than subsections, they will repeat and they're sort of obvious)
1.1 Terms
1.2 Semantics
1.3 Typing
2 [Simple proofs -- better name needed?]
2.1 Preservation of types: _structural induction_
2.2 Type uniqueness: _falsehood in proofs_
2.3 Termination of evaluation: _advanced structural induction?_
2.4 Small-step <-> big-step semantics: _explicit quantifiers_
(possibly also: 2.5 Progress of evaluation: _handling disjunctions_)
ALT (per the last bullet point)
2.1 Structural induction
2.1.1 Preservation of types
2.1.2 Termination of evaluation
2.2 Falsehood in proofs: type uniqueness
2.3 Explicit quantification: relating small-step and big-step semantics
(possibly also: 2.4 Handling disjunctions: progress of evaluation)
3. Trusting the proof environment

II. Towards a programming language
1. Simply-typed lambda calculus
1.1 Terms
1.2 Semantics
1.3 Typing
2. Encoding variables, binders and assumptions
2.1 De Bruijn indices
2.2 Higher-Order Abstract Syntax
3. [Proofs in the presence of binders  -- better name needed?]
3.1 Preservation of types: _.._
3.2  De Bruijn <-> HOAS translation: _.._
3.3 Type uniqueness: _.._

III. Language extensions and advanced proof techniques
(all the chapters that are not there yet; this might be split into 2 parts)

Another option is to keep a chapter-per-language(or extension) approach and be even more explicit about it (chapter title based on the language). In that case the parts on trusting the proof environment and de Bruijn vs HOAS are a bit odd to have as sections, as they are sort of "interludes", introducing slightly different concepts than the rest. A section explaining what LF is (if we decide to have one) will have a similar problem.

I'm curious to hear your thoughts and alternative suggestions :)

**Comment by pientka** [comment]
After talking to several of you, here are some comments:
- splitting the book up into different parts is a great idea; generally there was consensus to keep the idea of Basic, Intermediate, Advanced as headers for each part. 
- we should keep the names of theorems as section titles, but remove the subtitles after the colon. This is what most readers can identify with and will find meaningful. To make it easier to search for particular technqiues (i.e. how do I encode disjunctions?, existentials? falsehood?), I am suggesting to start an index where people can look up where such techniques and ideas are discussed.
  - Brigitte
